util/util.go

```diff
@@ -161,14 +161,36 @@ func GetReceipts(ctx context.Context, rawBlocks []*json.RawMessage, c *ethrpc.Cl
 			}
 			break
 		}
-
-		err := c.BatchCallContext(ctx, blms[start:end])
-		if err != nil {
-			log.Error().Err(err).Str("randtx", txHashes[0]).Uint64("start", start).Uint64("end", end).Msg("RPC issue fetching receipts")
-			return nil, err
-		}
-		start = end
-		if last {
+		// go-ethereum's BatchCallContext fails when the number of txns are over 100, because it attempts to unmarshal a single json source object into a slice if the number of txns go over 100.
```

Comment:
We already have a `batch-size` flag that should prevent this issue.
https://github.com/0xPolygon/polygon-cli/blob/main/cmd/loadtest/output.go#L318-L333

Also the limit you're talking about may be related to [erigon's rpc batch limit set to 100 by default](https://erigon.gitbook.io/erigon/advanced-usage/rpc-daemon).